app/src/org/commcare/android/framework/SessionAwareCommCareActivity.java

```diff
@@ -1,11 +1,30 @@
 package org.commcare.android.framework;
 
+import android.os.Bundle;
+
+import org.commcare.android.util.SessionUnavailableException;
+
 /**
  * Manage redirection to login screen when session expiration occurs.
  *
  * @author Phillip Mates (pmates@dimagi.com)
  */
 public abstract class SessionAwareCommCareActivity<R> extends CommCareActivity<R> {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        try {
+            onCreateAware(savedInstanceState);
+        } catch (SessionUnavailableException e) {
+            SessionActivityRegistration.redirectToLogin(this);
+        }
```

Comment:
Re-did the approach I linked to above to not call `finish()`, instead deferring to `onResume` where the user will be booted to the login screen via existing session aware logic (as far as I can tell)
https://github.com/dimagi/commcare-odk/commit/67467d121fe43acca18752acd35b3f5da436c3d2